### PR TITLE
feat(core): expose cmux in public API

### PR DESCRIPTION
**Pull request by Carl-Zama**
### Description

add fftw implementation, specification, fixture, test and benchmark for cmux

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (apply the `aws_test` to the PR to launch the tests on AWS)
* [x] The draft release description has been updated
* [x] Check for breaking changes and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Review by agnesLeroy — COMMENTED** [review]
Hi @Carl-Zama! Thanks a lot for this PR! Overall it looks great, I just have a few remarks here and there, nothing major. :slightly_smiling_face: Very good job!

File: concrete-core-fixture/src/fixture/glwe_ciphertexts_ggsw_ciphertext_fusing_cmux.rs [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+        let proto_secret_key =
+            maker.new_glwe_secret_key(parameters.glwe_dimension, parameters.polynomial_size);
```
[the remark this answers is not in the document]
Comment (Carl-Zama) [reply concrete-core-fixture/src/fixture/glwe_ciphertexts_ggsw_ciphertext_fusing_cmux.rs:108]: In the original tests there was a test where the GGSW plaintext was zero and a separate test for it being one, if I move GGSW ciphertext generation to be once per run should I also split the two cases or still just generate a random plaintext once?

File: concrete-core/src/specification/engines/glwe_ciphertexts_ggsw_ciphertext_fusing_cmux.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/// A trait for engines computing the controlled multiplexer (cmux) of two GLWE ciphertexts and a
+/// GGSW ciphertext.
+///
+/// # Semantics
+///
+/// This [fusing](super#operation-semantics) operation generates a GLWE ciphertext containing the
+/// result of the cmux between
+/// `glwe_input' and `glwe_ouput' GLWE ciphertexts and a `ggsw_input' GGSW ciphertext.
+///
+/// # Formal Definition
```
[the remark this answers is not in the document]
Comment (Carl-Zama) [reply concrete-core/src/specification/engines/glwe_ciphertexts_ggsw_ciphertext_fusing_cmux.rs:49]: I wrote something, can you check it is sufficient?

[the remark this answers is not in the document]
Comment (agnesLeroy) [reply concrete-core/src/specification/engines/glwe_ciphertexts_ggsw_ciphertext_fusing_cmux.rs:49]: Will do, I just sent a message in the concrete-framework-internal slack channel to ask for Ilaria's advice.

**Review by agnesLeroy — APPROVED** [review]

**Pull request merged**